### ISSUE TITLE
Additional lines added after writing brackets and pressing. <CR>

### ISSUE THREAD

**Issue by MarcoCamilo** [windwp/nvim-autopairs#412]
### Description

Every time I write a pair of brackets and press enter, the brackets are split but two lines are added and the cursor moves to the end of them. 

Expected:
```
Before        Input         After
------------------------------------
{|}           <CR>          {
                                |
                            }
------------------------------------
```

Behavior:
```
Before        Input         After
------------------------------------
{|}           <CR>          {
                            }

                            |
------------------------------------
```

### Mapping <cr> bug

1.If you report a bug about indent. Please remember that plugin doesn't do anything about indent.
  It just trigger the indent of your vim config so if you have wrong indent config then it will do wrong indent.
  You can check by select a block of code and press `==`
2. provide result of command `:verbose imap <cr>`.

### Minimal config

https://github.com/MarcoCamilo/nvim

**Comment by Sam-programs** [comment]
you didn't provide a minimal config with the issue
the default config doesn't have issues with `<CR>`
and your config seems to be using a really old version of nvim-autopairs
https://github.com/MarcoCamilo/nvim/blob/nvim_basic_ide/lua/user/plugins.lua#L49
maybe that's the issue

**Comment by MarcoCamilo** [comment]
Sorry, that wasn't my updated config. Now my actual config is available. However, I'm using the latest version of autopairs. I'm also using noice.nvim, and I noticed that in removing it, it kinda solves the problem, but I don't know why would it clash with autopairs.

**Comment by Sam-programs** [comment]
i am not really sure why noice.nvim does that with your config

your indenter might be the issue because autopairs doesn't control it
give me the value of indentexpr with noice.nvim
`:set indentexpr`

this is what the plugin is doing
```lua
<CR>
{|}

indent the end pair }
{
|}
maybe the indenter is moving the cursor down and adding a 2 lines?

up arrow key
{
|}

end key
|{
}

<CR>
{|
}

{
   |
}
```

**Comment by MarcoCamilo** [comment]
I hope I understood correctly what you asked me. The value I get with `:set indentexpr` is ` indentexpr=GetLuaIndent()`. But if it's not auto pair, then I don't know where to correct the behavior. 

As I mentioned, the behavior I'm seeing is that the cursor is being pushed two lines below the corresponding bracket pair when I press <CR>
```
{
}


|
```

**Comment by Sam-programs** [comment]
i finally figured it out!

tl;dr
when autopairs indents using == it sends a message to `noice.nvim`
`noice.nvim` handles that message while the `<cr>` mapping is going
u can prevent the message from being sent by setting 'report' to 1 or higher
https://github.com/MarcoCamilo/nvim/blob/master/lua/marco/core/options.lua#L24
since we are indenting 1 line anything more than 0 for report won't print see `:help 'report'`

i think this is why that behaviour was happening
```
<CR>
{|}

indent the end pair }
{
|}
noice handles the message from indenting

up arrow key
{
|}
the up arrow key movement doesn't happen (probably caused by noice)

end key
{
|}

<CR>
{
}|

{
}
|

then noice decides to insert an extra <CR> 
https://github.com/folke/noice.nvim/blob/main/lua/noice/ui/msg.lua#L131
{
}

|
```

**Comment by MarcoCamilo** [comment]
EUREKA! I can't believe the solution was so simple and not at all where I'd ever think it would be. Thank you so much for looking over my config and helping me with this!

**Comment by Sam-programs** [comment]
yeah i didn't come up with an unreadable solution before that totally...
lol
